tfworker/definitions.py

```diff
@@ -235,9 +237,19 @@ def iter(self, honor_destroy=False):
 
     def limited(self):
         # handle the case where nothing is filtered
-        if len(list(filter(lambda d: d.limited, self.iter(honor_destroy=True)))) == 0:
+        iter_size = len(
+            list(filter(lambda d: d.limited, self.iter(honor_destroy=True)))
+        )
+        if iter_size == 0:
+            # a limit was supplied, but not matched, raise an error
+            if self._limit:
+                raise ValueError("no items matching limit")
             # the run is not limited to anything, so return everything
-            return self.iter(honor_destroy=True)
+            else:
+                return self.iter(honor_destroy=True)
+        elif iter_size < self._limit_size:
+            # not all limit items are matched
+            raise ValueError("not all items matching limit")
```

Comment:
Are the `items` here definitions? Might be a more helpful error message to users to change wording to `definitions`.

Reply:
good call, updated both possible error messages to be more intuitive